py/embedded-server/deephaven_server/start_jvm.py

```diff
@@ -35,41 +36,78 @@ def _jars():
     'logback.configurationFile': 'logback-minimal.xml',
 }
 DEFAULT_JVM_ARGS = [
-    # Disable the JVM's signal handling for interactive python consoles - if python will
-    # not be handling signals like ctrl-c (for KeyboardInterrupt), this should be safe to
-    # remove for a small performance gain.
-    '-Xrs',
-
     # Disable JIT in certain cases
     '-XX:+UnlockDiagnosticVMOptions',
     f"-XX:CompilerDirectivesFile={_compiler_directives()}",
     f"-XX:VMOptionsFile={_default_vmoptions()}",
 ]
 
+def _merge_with_default_jvm_args(jvm_args: Optional[List[str]] = None) -> List[str]:
+    """ Merge the default JVM args with the user-provided JVM args."""
+    if not jvm_args:
+        return DEFAULT_JVM_ARGS
+
+    result_args = DEFAULT_JVM_ARGS.copy()
+
+    # Remove any conflicting options
+    for jvm_arg in jvm_args:
```

Comment:
I'm worried about the complexity of this logic and maintaining it long term.

I'd like to get others thoughts on two sets of jvm arguments like the native script has. `JAVA_OPTS` is set by default, and only recommended to override for more advanced use cases. `START_OPTS` is recommended way to set most args. And then, the start script essentially just passes `JAVA_OPTS` + `START_OPTS` to the JVM. In this case, something like:

```python
start_jvm(
  default_jvm_args: Optional[List[str]] = None,
  jvm_args: Optional[List[str]] = None,
  ...
)
```

with documentation recommending most users only set jvm_args.

Reply:
I am wary of introducing breaking changes but I agree that the separation of two types of JVM args makes the code cleaner and more maintainable. Something like this would serve both goals. Any thoughts?
```
def __init__(
        self,
        host: Optional[str] = None,
        port: Optional[int] = None,
        jvm_args: Optional[List[str]] = None,
        extra_classpath: Optional[List[str]] = None,
        default_jvm_args: Optional[List[str]] = None,
    ):
```

Reply:
Yes, I think something like that would be much better